### PR TITLE
VSCode | Fix UI Freezing and Unclickable Icons (AST-70028)

### PR DESCRIPTION
**Pull request by AlvoBen**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> Fix bug of unclickable icons - give user indication that their apikey is invalid.

### References

> https://checkmarx.atlassian.net/browse/AST-70028

### Testing

> X

### Checklist

- [x] I have added documentation for new/changed functionality in this PR (if applicable).
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

File: src/utils/listener/listeners.ts [hunk 1]
```diff
@@ -179,7 +179,7 @@ export async function executeCheckSettingsChange(
     );
     vscode.commands.executeCommand(
       commands.setContext,
-      commands.isValidCredentials,
+      commands.isScanEnabled,
```
Comment (miryamfoiferCX) [inline src/utils/listener/listeners.ts:182]: you just renamed?
or use another parameter? - if use another - do we still need to save the old parameter somewhere?

File: src/cx/cx.ts [hunk 2]
```diff
@@ -253,7 +253,9 @@ export class Cx implements CxPlatform {
 		try {
 			enabled = await cx.ideScansEnabled();
 		} catch (error) {
-			logs.error(error);
+			const errMsg = `Error checking tenant configuration: ${error}`;
+			vscode.window.showErrorMessage("Provided Checkmarx One API KEY is invalid");
```
[the remark this answers is not in the document]
Comment (AlvoBen) [reply src/cx/cx.ts:257]: Yes, I need to talk about it with Daniel

File: src/cx/cx.ts [hunk 3]
```diff
@@ -253,7 +253,9 @@ export class Cx implements CxPlatform {
...
+            logs.error(errMsg);
 			return enabled;
```
[the remark this answers is not in the document]
Comment (AlvoBen) [reply src/cx/cx.ts:259]: enable will be false

File: src/cx/cx.ts [hunk 4]
```diff
@@ -1,456 +1,483 @@
...
+                logs.info(authSuccessMsg);
+                vscode.window.showInformationMessage(authSuccessMsg);
+                return true;
+            } else {
+                logs.error(`${invalidAPIKeyMsg}: ${valid.status}`);
+                vscode.window.showErrorMessage(invalidAPIKeyMsg);
+                return false;
+            }
+        } catch (error) {
+            logs.error(`${invalidAPIKeyMsg}: ${error}`);
```
Comment (miryamfoiferCX) [inline src/cx/cx.ts:476]: if we got exit code != 0, it means that apikey is invalid
but if we catch an exception it means other issue happened, could be not related to apikey itself
no?

File: src/cx/cx.ts [hunk 5]
```diff
@@ -1,456 +1,483 @@
...
+            vscode.window.showErrorMessage(invalidAPIKeyMsg);
```
Comment (miryamfoiferCX) [inline src/cx/cx.ts:477]: same

File: src/utils/listener/listeners.ts [hunk 6]
```diff
@@ -1,203 +1,223 @@
...
+                : messages.kicsStatusBarDisconnect;
+        await vscode.commands.executeCommand(commands.refreshTree);
+        const ascaEffected = event.affectsConfiguration(
+            `${constants.CheckmarxAsca}.${constants.ActivateAscaAutoScanning}`
+        );
+        const apikeyEffected = event.affectsConfiguration(
+            "checkmarxOne.apiKey"
+        );
+        
+        if (apikeyEffected) {
```
Comment (miryamfoiferCX) [inline src/utils/listener/listeners.ts:215]: do we have a test for this case?

File: src/cx/cx.ts [hunk 7]
```diff
@@ -1,456 +1,483 @@
...
+        const cx = new CxWrapper(config);
+        const scans = await cx.scanAsca(sourcePath, false, constants.vsCodeAgent);
+        if (scans.payload && scans.exitCode === 0) {
+            return scans.payload[0];
+        } else {
+            return this.getAscaError(scans.status, "Fail to call ASCA scan");
+        }
+    }
+
+    async authValidate(logs: Logs): Promise<boolean> {
```
Comment (miryamfoiferCX) [inline src/cx/cx.ts:459]: do we have a test to this logic?

**Review by miryamfoiferCX — APPROVED** [review]

**Pull request merged**